substrate/frame/support/src/traits/misc.rs

```diff
@@ -48,6 +48,10 @@ impl VariantCount for () {
 	const VARIANT_COUNT: u32 = 0;
 }
 
+impl VariantCount for u8 {
+	const VARIANT_COUNT: u32 = 256;
+}
```

Comment:
I think this change itself is fine. If people opt to (ab)use `u8` for Hold or Freeze reason is pretty independant of this.